### PR TITLE
Add Buffer Balances to Wrap/Unwrap and Buffer liquidity events

### LINKED ISSUE #117: Address V2 Pain Points from Data Team
https://www.notion.so/V2-Pain-Points-a66a412d7a4644b8854467c45aabef37?pvs=4

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

The Data team needs buffer balances in these events so they can properly measure the buffer interactions.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #117 
This is "addresses pain points from the SDK team." Is this the last such thing?

### CODE REVIEW

File: pkg/vault/contracts/Vault.sol [hunk 1]
```diff
@@ -1171,25 +1182,21 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
...
                 // We have verified above that this is safe to do unchecked.
-                newDerivedBalance = bufferBalances.getBalanceDerived() - amountOutWrapped;
+                bufferWrappedBalance = bufferBalances.getBalanceDerived() - amountOutWrapped;
             }
-
-            bufferBalances = PackedTokenBalance.toPackedBalance(
-                bufferBalances.getBalanceRaw() + amountInUnderlying,
-                newDerivedBalance
-            );
+            bufferBalances = PackedTokenBalance.toPackedBalance(bufferUnderlyingBalance, bufferWrappedBalance);
```
Comment (jubeira) [inline pkg/vault/contracts/Vault.sol:1199]: I mean this looks good in general, but the stakes are too high at this point. 

I don't feel comfortable modifying all of these lines just to modify a few fields in an event.

How about just returning the packed balances and unpacking outside? Maybe not the most efficient, but you won't need any extra (warm) storage reads and the diff will be even smaller.

**Review by jubeira — COMMENTED** [review]
Thanks for pushing this @joaobrunoah !

The changes look good, but I'm a bit more reluctant than usual with things like these. Let me think a bit more about it.

File: pkg/vault/contracts/Vault.sol (hunk 1 above)
Comment (jubeira) [reply pkg/vault/contracts/Vault.sol:1199]: The function now returns 4 arguments, there are multiple if / else levels, early returns, etc. If we stare too much at the function we might break it.

**Comment by joaobrunoah** [comment]
Superseded by #1107

**Pull request closed without merging**